src/test/java/core/basesyntax/strategy/SupplyDataHandlerTest.java

```diff
@@ -0,0 +1,70 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.data.FruitTransaction;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class SupplyDataHandlerTest {
+    private static final int QUANTITY_BANANA = 75;
+    private static final String FRUIT_BANANA = "banana";
+    private SupplyDataHandler supplyDataHandler;
+    private FruitTransaction validTransaction;
+    private FruitTransaction invalidTransaction;
+    private Map<String, Integer> validData;
+
+    @BeforeEach
+    void setUp() {
+        supplyDataHandler = new SupplyDataHandler();
+    }
+
+    @Test
+    void validPurchaseOperationInTransaction_Ok() {
+        validTransaction =
+                new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        FRUIT_BANANA, QUANTITY_BANANA);
+        Assertions.assertTrue(validTransaction.getOperation()
+                .equals(FruitTransaction.Operation.BALANCE));
+    }
+
+    @Test
+    void invalidPurchaseOperationInTransaction_notOk() {
+        invalidTransaction =
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                        FRUIT_BANANA, QUANTITY_BANANA);
+        Assertions.assertFalse(FruitTransaction.Operation.BALANCE
+                .equals(invalidTransaction.getOperation()));
+    }
+
+    @Test
+    void validProcessWithData_Ok() {
+        validData = Map.of(
+                FRUIT_BANANA,QUANTITY_BANANA);
+        validTransaction =
+                new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        FRUIT_BANANA, QUANTITY_BANANA);
+        Assertions.assertTrue(
+                validData.containsKey(validTransaction.getFruit()));

```

Comment:
let's better use for this test `assertEquals` on `FRUIT_BANANA` and `validTransaction.getFruit()`. Because now you also test `containsKey` method.